lib/shopify_app/login_protection.rb

```diff
@@ -27,7 +27,7 @@ def shop_session
     end
 
     def login_again_if_different_shop
-      if shop_session && params[:shop] && params[:shop].is_a?(String) && shop_session.url != params[:shop]
+      if shop_session && params[:shop] && (shop_session.url != params[:shop])
```

Comment:
I generally like extracting logic like this into well-named methods, but I don't think that we should do that in this case. The logic here is easy to understand because `#login_again_if_different_shop` gives us enough context.

I have been thinking about the `LoginProtection` module and I am a little worried about the methods that are exposed to the public API because they are `protected` and not `private`. I don't have a good solution for this yet but we should think about how to reduce the public API of this gem.